modules/account-lib/src/coin/stx/transferBuilder.ts

```diff
@@ -0,0 +1,103 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import BigNum from 'bn.js';
+import {
+  makeUnsignedSTXTokenTransfer,
+  PayloadType,
+  UnsignedTokenTransferOptions,
+  TokenTransferOptions,
+  UnsignedMultiSigTokenTransferOptions,
+} from '@stacks/transactions';
+import { TransactionType } from '../baseCoin';
+import { InvalidParameterValueError, InvalidTransactionError, BuildTransactionError } from '../baseCoin/errors';
+import { Transaction } from './transaction';
+import { TransactionBuilder } from './transactionBuilder';
+import { isValidAddress, isValidAmount, removeHexPrefix } from './utils';
+
+export class TransferBuilder extends TransactionBuilder {
+  private _options: UnsignedTokenTransferOptions | UnsignedMultiSigTokenTransferOptions;
+  private _toAddress: string;
+  private _amount: BigNum;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  initBuilder(tx: Transaction): void {
+    const txData = tx.toJson();
+    if (txData.payload === undefined) {
+      throw new InvalidTransactionError('payload must not be undefined');
+    }
+    if (txData.payload.payloadType == PayloadType.TokenTransfer) {
+      this.to(txData.payload.to!);
+      this.amount(txData.payload.amount!);
+      super.initBuilder(tx);
+    } else {
+      throw new BuildTransactionError("Transaction should be transfer")
+    }
+
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this._options = this.buildTokenTransferOptions();
+    this.transaction.setTransactionType(TransactionType.Send);
+    this.transaction.stxTransaction = await makeUnsignedSTXTokenTransfer(this._options);
+    return await super.buildImplementation();
+  }
+
+  private buildTokenTransferOptions(): UnsignedTokenTransferOptions | UnsignedMultiSigTokenTransferOptions {
+    const defaultOpts: TokenTransferOptions = {
+      recipient: this._toAddress,
+      amount: this._amount,
+      memo: this._memo,
+      network: this._network,
+      fee: new BigNum(this._fee.fee),
+      nonce: new BigNum(this._nonce),
+    };
+    if (this._multiSignerKeyPairs.length === 0) {
+      throw new InvalidParameterValueError('supply at least 1 public key');
+    }
+    if (this._multiSignerKeyPairs.length === 1) {
+      return {
+        ...defaultOpts,
+        publicKey: this._multiSignerKeyPairs[0].getKeys(this._multiSignerKeyPairs[0].getCompressed()).pub,
+      };
```

Comment:
How come we need this special path for single keypairs?

Reply:
There is difference in keys for single and multi sig tx. (`publicKey` vs `publicKeys`)